kroxylicious-operator/src/test/java/io/kroxylicious/kubernetes/operator/VirtualKafkaClusterReconcilerTest.java

```diff
@@ -55,55 +75,120 @@ class VirtualKafkaClusterReconcilerTest {
+                        CLUSTER_NO_FILTERS,
+                        Optional.of(PROXY),
+                        Optional.empty(),
+                        Stream.of(INGRESS),
+                        Stream.of(),
+                        (Consumer<ConditionAssert>) ca -> ca.isResolvedRefsFalse(
+                                "ReferencedResourcesNotFound",
+                                "spec.targetKafkaServiceRef references kafkaservice.kroxylicious.io/my-kafka")),
+                Arguments.argumentSet("ingress not found",
+                        CLUSTER_NO_FILTERS,
+                        Optional.of(PROXY),
+                        Optional.of(SERVICE),
+                        Stream.of(),
+                        Stream.of(),
+                        (Consumer<ConditionAssert>) ca -> ca.isResolvedRefsFalse(
+                                "ReferencedResourcesNotFound",
+                                "spec.ingressRefs references kafkaproxyingress.kroxylicious.io/my-ingress")),
+                Arguments.argumentSet("filter not found",
+                        CLUSTER_ONE_FILTER,
+                        Optional.of(PROXY),
+                        Optional.of(SERVICE),
+                        Stream.of(INGRESS),
+                        Stream.of(),
+                        (Consumer<ConditionAssert>) ca -> ca.isResolvedRefsFalse(
+                                "ReferencedResourcesNotFound",
+                                "spec.filterRefs references kafkaprotocolfilter.filter.kroxylicious.io/my-filter")));
     }
 
-    @Test
-    void shouldSetResolvedRefsToTrueWhenProxyFound() {
+    @ParameterizedTest
+    @MethodSource
+    void shouldSetResolvedRefsToTrueOrFalse(VirtualKafkaCluster cluster,
+                                            Optional<KafkaProxy> existingProxy,
+                                            Optional<KafkaService> existingService,
+                                            Stream<KafkaProxyIngress> existingIngresses,
+                                            Stream<KafkaProtocolFilter> existingFilters,
+                                            Consumer<ConditionAssert> asserter) {
         // given
         Clock z = Clock.fixed(Instant.EPOCH, ZoneId.of("Z"));
         var reconciler = new VirtualKafkaClusterReconciler(z);
 
         Context<VirtualKafkaCluster> context = mock(Context.class);
-        when(context.getSecondaryResource(KafkaProxy.class, VirtualKafkaClusterReconciler.PROXY_EVENT_SOURCE_NAME)).thenReturn(Optional.of(PROXY));
+        when(context.getSecondaryResource(KafkaProxy.class, VirtualKafkaClusterReconciler.PROXY_EVENT_SOURCE_NAME)).thenReturn(existingProxy);
+        when(context.getSecondaryResource(KafkaService.class, VirtualKafkaClusterReconciler.SERVICES_EVENT_SOURCE_NAME)).thenReturn(existingService);
+        when(context.getSecondaryResourcesAsStream(KafkaProxyIngress.class)).thenReturn(existingIngresses);
```

Comment:
wondering if these stream mocks are a bit brittle, if we ever wanted to call them twice in future the stream would have been consumed.